### PR TITLE
[Merged by Bors] - refactor(PartialHomeomorph): Remove explicit variable (x : M).

### PR DESCRIPTION
**Pull request by grunweg**
Instead, supply it as needed.

This replaces an explicit argument (x : M) by an implicit `{x : M}` in the following lemmas:
- extChartAt_source_mem_nhds'
- extChartAt_source_mem_nhdsWithin'
- continuousAt_extChartAt'
- extChartAt_image_nhd_mem_nhds_of_boundaryless
- extChartAt_target_mem_nhdsWithin'
- nhdsWithin_extChartAt_target_eq'
- continuousAt_extChartAt_symm'
- continuousAt_extChartAt_symm''
- map_extChartAt_nhdsWithin_eq_image'
- map_extChartAt_nhdsWithin'
- map_extChartAt_symm_nhdsWithin'
- map_extChartAt_symm_nhdsWithin_range'
- extChartAt_preimage_mem_nhdsWithin'
- extChartAt_preimage_mem_nhdsWithin
- extChartAt_preimage_mem_nhds'
- extChartAt_preimage_mem_nhds

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean [hunk 1]
```diff
@@ -1316,192 +1316,195 @@ theorem uniqueDiffWithinAt_extChartAt_target (x : M) :
...
-    (extChartAt I x).symm ⁻¹' t ∈ 𝓝 (extChartAt I x x') :=
+theorem extChartAt_preimage_mem_nhds' {x x' : M} (h : x' ∈ (extChartAt I x).source)
+    (ht : t ∈ 𝓝 x') : (extChartAt I x).symm ⁻¹' t ∈ 𝓝 (extChartAt I x x') :=
   extend_preimage_mem_nhds _ _ (by rwa [← extChartAt_source I]) ht
 #align ext_chart_at_preimage_mem_nhds' extChartAt_preimage_mem_nhds'
 
 /-- Technical lemma ensuring that the preimage under an extended chart of a neighborhood of a point
 is a neighborhood of the preimage. -/
-theorem extChartAt_preimage_mem_nhds (ht : t ∈ 𝓝 x) :
+theorem extChartAt_preimage_mem_nhds (x : M) (ht : t ∈ 𝓝 x) :
```
Comment (winstonyin) [inline Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean:1499]: Same here. Did you mean:
```suggestion
theorem extChartAt_preimage_mem_nhds {x : M} (ht : t ∈ 𝓝 x) :
```

**Comment by grunweg** [comment]
@winstonyin Thank you for the sharp eye. You were right.

File: Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean [hunk 2]
```diff
@@ -1318,193 +1318,196 @@ theorem uniqueDiffWithinAt_extChartAt_target (x : M) :
...
 
-theorem nhdsWithin_extChartAt_target_eq :
+theorem nhdsWithin_extChartAt_target_eq (x : M) :
     𝓝[(extChartAt I x).target] (extChartAt I x) x = 𝓝[range I] (extChartAt I x) x :=
-  nhdsWithin_extChartAt_target_eq' I x (mem_extChartAt_source I x)
+  nhdsWithin_extChartAt_target_eq' I (mem_extChartAt_source I x)
 #align nhds_within_ext_chart_at_target_eq nhdsWithin_extChartAt_target_eq
 
-theorem continuousAt_extChartAt_symm'' {y : E} (h : y ∈ (extChartAt I x).target) :
+theorem continuousAt_extChartAt_symm'' (x : M) {y : E} (h : y ∈ (extChartAt I x).target) :
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean:1405]:
```suggestion
theorem continuousAt_extChartAt_symm'' {x : M} {y : E} (h : y ∈ (extChartAt I x).target) :
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by grunweg** [comment]
Thank you for the review.
bors r+

**Pull request closed without merging**